docs/enhancements/backup-restore-support.md

```diff
@@ -0,0 +1,151 @@
+
+#### Story 4
+
+As an ACM/OADP or ZTP user, I want to restore/move a spoke cluster in non-installed status (draft/in-progress/etc). The cluster on target hub should, ideally, keep the same status. Or, at least, fail gracefully.
+
+### Implementation Details/Notes/Constraints
+
+#### Solution for BMHs reconcile (due to a missing status)
+
+According to BMO documentation, the suggested approach for moving BareMetalHost from one cluster to another is using the `baremetalhost.metal3.io/status` [annotation](https://github.com/metal3-io/metal3-docs/blob/main/docs/user-guide/src/bmo/status_annotation.md).
+I.e. Extract the current status of the BMH, store it in an annotation, create the BMH on the new hub:
+```
+# Save the status in json format to a file
+kubectl get bmh <name-of-bmh> -o jsonpath="{.status}" > status.json
+# Save the BMH and apply the status annotation to the saved BMH.
+kubectl -n metal3 annotate bmh <name-of-bmh> \
+  baremetalhost.metal3.io/status="$(cat status.json)" \
+  --dry-run=client -o yaml > bmh.yaml
+```
+This flow ensures that the BMH on the new hub has status (BMO applies the status using the annotation). Hence, as the BMH should be in state 'Provisioned', the BMH reconcile is avoided.
+I.e. prevent destruction of restored spoke clusters.
+
+##### Suggested solution for automating the flow (make it transparent to the user):
+
+For BMHs in the active hub: (In assisted-service -> `bmh_agent_controller`)
+* Once the BMH is in 'Provisioned' state:
+  * Extract BMH's status and add `baremetalhost.metal3.io/status` to the object metadata.
+  * Add also, add the `baremetalhost.metal3.io/paused` annotation:
+    * This is required since once the annotation is added, BMO updates the BMH's status and delete the annotation.
+    * The [paused](https://github.com/metal3-io/baremetal-operator/blob/main/docs/api.md#pausing-reconciliation) annotation, is similar to the `detached` annotation we're already using - but, it's also pausing the reconcile loop. Thus, keeping the status annotation intact.
+
+For BMHs in the passive hub: (In assisted-service -> `bmh_agent_controller`)
+* When the BMH is applied, the status will be missing - since reconcile is paused.
+  * To restore the original status:
+    * Remove the `baremetalhost.metal3.io/paused` annotation
+      * Status is restored, but `baremetalhost.metal3.io/status` annotation is removed.
+    * Hence, add the status and paused annotation back
+      * By detecting a BMH with `provisioned` state and a missing status annotation.  
+
+#### Solution for deleted Agent CRs
+
+1. Don't delete the Agent CR when an associated host doesn't exits on DB.
+   * Remove relevant [code](https://github.com/openshift/assisted-service/blob/5bbf696fc0cc5a220fd8a0d1e408ab0bcc889e3f/internal/controller/controllers/agent_controller.go#L143-L151) from AgentController.
```

Comment:
Are we ever relying on this code to delete agents for us in other cases?

For example if a user deletes an infraenv, does that delete the hosts internally and rely on this to remove the agent CRs or do we have finalizer logic that does this all already?

IIRC when a cluster with an infraenv bound directly to it (early binding) is removed, we should also remove the agents in that cluster.

Reply:
Yup, that looks like it deletes the CR which would allow the agent controller to do the delete for the db record.